### ISSUE TITLE
Concat decorator implementation appears wrong

### ISSUE THREAD

**Issue by codeandcats** [joiful-ts/joiful#28]
I was poking around the any constraints and noticed that `Concat` looks like it's possibly implemented incorrectly. As you can see `schema` is declared twice.

```typescript
/**
 * Returns a new type that is the result of adding the rules of one type to another.
 */
export function Concat(schema : Schema) : PropertyDecorator {
    return constraintDecorator([], (schema : Schema) => {
        return schema.concat(schema);
    });
}
```

I suspect this should be:

```typescript
/**
 * Returns a new type that is the result of adding the rules of one type to another.
 */
export function Concat(schema : Schema) : PropertyDecorator {
    return constraintDecorator([], (existingSchema : Schema) => {
        return existingSchema.concat(schema);
    });
}
```

Empty seems to be the same:

```typescript
/**
 * Outputs the original untouched value instead of the casted value.
 */
export function Empty(schema : any) : PropertyDecorator {
    return constraintDecorator([], (schema : Schema) => {
        return schema.empty(schema);
    });
}
```

@laurence-myers if you agree I'm happy to create a PR with the fix.

**Comment by laurence-myers** [comment]
Good catch! PR would be welcome.

**Comment by laurence-myers** [comment]
Fixed in [v0.0.12](https://github.com/laurence-myers/tsdv-joi/releases/tag/v0.0.12).
